.github/workflows/clean-aws-resources.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: Monthly Resources Cleaner
+
+on:
+  schedule:
+    - cron: "0 0 1 * *" # Run on the first day of every month
+  workflow_dispatch:
+    inputs:
+      daystokeep:
+        description: "days to keep a resource"
+        required: false
+        default: "30"
+
+env:
+  DAYS_TO_KEEP: 30
+
+jobs:
+  clean-resources:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        resource: [ami, dedicated_host, ssm]
+        region: [ us-east-1, us-east-2, us-west-2, eu-west-1 ]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.TERRAFORM_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.TERRAFORM_AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ matrix.region }}
+
+      - name: Update days to keep
+        if: github.event_name == 'workflow_dispatch'
+        run: echo "DAYS_TO_KEEP=${{ github.event.inputs.daystokeep }}" >> $GITHUB_ENV
```

Comment:
Are you sure this works? I'm not sure that piping to the `$GITHUB_ENV` would do anything. Is there any documentation that points to doing this

Reply:
Yes I am sure this would work. Here is a document regarding how to [set an environment variable using bash script](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable). Moreover,